lib/netconf/provider.go

```diff
@@ -113,6 +163,40 @@ func networkFromPortals(ctx context.Context, network ID, portals []bindings.Port
 	}, nil
 }
 
+// TODO: Combine & de-dup with method above.
+func networkFromRtypesPortals(ctx context.Context, network ID, portals []*rtypes.Portal) Network {
+	namer := ChainNamer(network)
+
+	var chains []Chain
+	for _, portal := range portals {
+		// Ephemeral networks may contain mock portals for testing purposes, just ignore them.
+		if _, ok := evmchain.MetadataByID(portal.ChainId); !ok && network.IsEphemeral() {
+			log.Warn(ctx, "Ignoring epheral network mock portal", nil, "chain_id", portal.ChainId)
+			continue
+		}
+
+		chains = append(chains, Chain{
+			ID:            portal.ChainId,
+			Name:          namer(portal.ChainId),
+			PortalAddress: common.BytesToAddress(portal.GetAddress()),
+			DeployHeight:  portal.DeployHeight,
+			// TODO: How can I get this info?
+			BlockPeriod: 0,
+			Shards:      toShardIDs(portal.GetShardIds()),
+			// TODO: How can I get this info?
```

Comment:
@corverroos Any ideas here?

Reply:
ah damn, when we added there fields to the execution portal registry contract, [PR](https://github.com/omni-network/omni/pull/1469), we forgot to also add them to the consensus portal registry. We should add them since they are consensus critical 🤦